rows.go

```diff
@@ -156,15 +178,17 @@ func (x *XlsxFile) parseRawCells(rawCells []rawCell, index int) ([]Cell, error)
 			continue
 		}
 		column := strings.Map(removeNonAlpha, rawCell.Reference)
-		val, err := x.getCellValue(rawCell)
+		val, typ, err := x.getCellValue(rawCell)
```

Comment:
Again - might read better doing 
```
val, err := x.getCellValue(rawCell) 
cellType, err := x.getCellType(rawCell)
```

?

Reply:
Might even be able to derive the type entirely from the cell, and not need to pass `x` - `rawCell.type()` ?

Reply:
The problem was `convertExcelDateToDateString()`, called in `getCellValue()`. That decides if something is a date or a datetime, and that trickles up.

The solution I went for was dropping the `TypeDate` class. It's a type excel doesn't really have anyway. This now makes parsing a formatted datetime cell to reformat it a bit more difficult (you have to first try to parse it with `2006-01-02`, and if that doesn't work, parse it as an time.RFC3339), but it de-tangles the code very nicely. If we add a `cell.AsTime()` function, as you suggested before, even the parsing problem is solved.